File: python/mandelbrot.py
import numpy as np
from pathlib import Path
side = 1024
center = .4+.4j
scale = 0.3
i_n = 512
x = np.linspace(-scale,scale,side)
y = np.linspace(-scale,scale,side, dtype=complex) * 1j
c = np.tile(x, (side,1))+np.tile(y, (side, 1)).T + center
z = np.zeros((side,side))
r = np.zeros((side,side),dtype='uint16')
for k in range(i_n):
    z = z ** 2 + c
    r[np.abs(z) < 2] = k


image = np.tile((1 - r.flatten()  / i_n) * 255, (3, 1)).astype('uint8').tobytes('F')
Path('out.ppm').write_bytes(f'P6 {side} {side} 255 '.encode() + image)

File: julia/mandelbrot.jl
const i_n = 512
const side = 1024
const center = 0.4+0.4im
const scale = 0.3

const xs = range(-scale, scale, length=side)
const ys = range(-scale, scale, length=side) .* 1im

const r = [0 for _ in 1:side*side]

let
    c = [x + y for y in ys for x in xs] .+ center
    z = zeros(side*side)
    for i in 1:i_n
        z = z .* z .+ c
        r[abs.(z) .< 2] .= i
    end
end
bytes = Vector{UInt8}("P6 $side $side 255 ")
const arr = Vector{UInt8}( floor.(255 .- (r ./ i_n .* 255)))
const content = reshape(transpose(repeat(arr, outer=[1, 3])), (side*side*3,))
append!(bytes, content)
open("out.ppm", "w") do io
    write(io, bytes)
end
